### PR TITLE
Add env vars for config paths

### LINKED ISSUE #47: Feature Request - Support environment variables in configuration paths
Hey 👋🏾 

I would like to ask if it would be possible to allow using environment variables within the configuration file. Especially for the paths. I want to be able to do something like this:

```json
{
  "installation_location": "$XDG_DATA_HOME/bob/bin"
}
```

Is that hard to do in regards of multi OS support?

### PR DESCRIPTION
**Pull request by MordechaiHadad**
fixes #47

### CODE REVIEW

**Comment by MordechaiHadad** [comment]
@weilbith can you test this? (i dont expect it to work right away)

**Comment by MordechaiHadad** [comment]
From my testing it seems like it works.

File: src/main.rs [hunk 1]
```diff
@@ -26,16 +27,50 @@ async fn main() -> Result<()> {
...
+    if let Some(value) = &config.installation_location {
+        let new_value = handle_envar(value, &re)?;
+        config.installation_location = Some(new_value);
+    }
+
     Ok(())
 }
+
+fn handle_envar(value: &str, re: &Regex) -> Result<String> {
+    let extract = re.captures(value).unwrap().get(1).unwrap().as_str();
```
Comment (Avishayy) [inline src/main.rs:71]: this breaks regular config usage, if the value doesn't conform to the regex, we panic here

Comment (MordechaiHadad) [reply src/main.rs:75]: fixed, thank you for noticing

**Comment by weilbith** [comment]
Sorry for the delay. I somehow missed the first notification.
So I checked-out this branch, build it and adjusted my configuration. Works fine. 👍🏾

**Comment by MordechaiHadad** [comment]
> Sorry for the delay. I somehow missed the first notification. So I checked-out this branch, build it and adjusted my configuration. Works fine. 👍🏾 Thank you very much!

Thank you for testing!

**Pull request merged**